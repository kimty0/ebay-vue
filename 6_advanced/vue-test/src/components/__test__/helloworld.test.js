import { mount, shallowMount } from '@vue/test-utils'
import HelloWorld from '../HelloWorld.vue'

describe('Hello World Component', () => {
  // component testing
  test('[Component Testing] message equals to be Hello Vue.js', () => {
    const { vm } = shallowMount(HelloWorld);
    expect(vm.message).toBe('Hello Vue.js');
  });

  // snapshot testing
  test('[Snapshot Testing] renders hello world message', () => {
    const { vm } = mount(HelloWorld);
    expect(vm.$el).toMatchSnapshot();
  });
});