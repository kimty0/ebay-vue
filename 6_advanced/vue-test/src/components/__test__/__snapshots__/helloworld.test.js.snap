// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World Component [Snapshot Testing] renders hello world message 1`] = `
<div>
  
  Hello Vue.js

</div>
`;
